### ISSUE TITLE
test not working 

### ISSUE THREAD

**Issue by vifeng** [vueschool/A-Wordle-on-Test-Driven-Development-in-Vue.js#1]
Bug :
Both this commit treat of a test that is not passing.
commit : 1361493384048f1674e332afb5725ab300a143e0
commit : eceb0b1d7e1afbb7124fdd7813ca332e7b84db90

```
 test('non-letter characters do not render on the screen while being typed', async () => {
      await playerSubmitsGuess('12')
      await playerSubmitsGuess('123')

      expect(wrapper.find<HTMLInputElement>('input[type=text]').element.value).toEqual('')
    })
```
Reason : 
no idea

Solution :
In the meantime, just skip `test.skip('non-letter....`

**Comment by SheepNDW** [comment]
To ensure tests pass, you must force the `computed` to update. Here's my solution:

```ts
const formattedGuessInProgress = computed({
  get: () => guessInProgress.value,
  set: (rawValue: string) => {
    const formattedValue = rawValue
      .slice(0, WORD_SIZE)
      .toUpperCase()
      .replace(/[^A-Z]+/gi, '');

    guessInProgress.value = formattedValue;
    // need force update to trigger reactivity
    triggerRef(formattedGuessInProgress);
  }
});
```

This is because Vue 3.4 made some optimizations to computed, as referenced here: https://vuejs.org/guide/best-practices/performance.html#computed-stability

**Comment by FRFlor** [comment]
Thanks for the suggestion @SheepNDW 

At the time that I recorded the course, I was using Vue 3.3,

As you mentioned in your post, using triggerRef() does indeed fix this issue for Vue 3.4.

Thanks again for sharing that, I'll reach out to the other members of VueSchool to see how we can update the course to let others know about this.

**Comment by lordvidex** [comment]
Hello, I have tried the above solutions, but my tests are still not passing. `triggerRef` doesn't seem to fix the issue for Vue 3.5.13

**Comment by AdriBarda** [comment]
Hey guys 👋
I’ve got a question about the solution you suggested.
Wouldn’t it be simpler to extract the sanitise logic into a separate function and just watch the guessInProgress ref?
Do you see any downsides to this approach or any caveats I might be missing?
I’ll drop a code example below ⬇️  Currently trying it out on vue 3.5.22

```js 
const sanitise = (value: string) => {
  return value
    .slice(0, WORD_SIZE)
    .toUpperCase()
    .replace(/[^A-Z]/g, '')
}
watch(guessInProgress, (raw) => {
  const clean = sanitise(raw)
  if (clean !== raw) {
    guessInProgress.value = clean
  }
})
```
